### PR TITLE
Introduce rejected snapshots

### PR DESCRIPTION
**Pull request by cadeParade**
## Description
This PR adds the rejected properties to snapshot + build model, and also allows the build to fetch snapshots in state rejected.

This PR does not add any different display based on snapshot state rejected, nor does it sort the snapshots on the build page differently yet.

## Reviewing this PR
_**Is Covered by Automated Tests?: YES**_
I've added a couple of rejected snapshots to tests that should test the order of snapshots. All the tests passing and no visual diffs related to snapshot ordering means that we successfully fetch rejected snapshots and do not sort them differently.

### CODE REVIEW

File: app/models/snapshot.js [hunk 1]
```diff
@@ -53,12 +59,18 @@ export default DS.Model.extend({
...
   // - 'approved' --> 'auto_approved_branch': Automatically approved based on branch name
+  // - 'rejected' --> 'user_rejected': User clicked "Request changes" button in current build
+  // - 'rejected' --> 'user_rejected_previously':'rejected' status has carried forward
   reviewStateReason: DS.attr(),
   isApprovedByUser: equal('reviewStateReason', SNAPSHOT_REVIEW_STATE_REASONS.USER_APPROVED),
   isApprovedByUserPreviously: equal(
     'reviewStateReason',
     SNAPSHOT_REVIEW_STATE_REASONS.USER_APPROVED_PREVIOUSLY,
   ),
+  isRejectedPreviously: equal(
```
Comment (unicornzero) [inline app/models/snapshot.js]: Should there also be an `isUserRejected` mapping to the `user_rejected` reviewState?

File: tests/factories/comment-thread.js [hunk 2]
```diff
@@ -32,5 +32,10 @@ FactoryGuy.define('comment-thread', {
     note: {
       type: NOTE_COMMENT_TYPE,
     },
+
+    autoReview: {
+      type: REVIEW_COMMENT_TYPE,
+      comments: FactoryGuy.hasMany('comment', 1, {body: 'Changes were requested'}),
```
Comment (unicornzero) [inline tests/factories/comment-thread.js]: Might want to make the body blank later.

Comment (unicornzero) [reply tests/factories/comment-thread.js]: (To match what is happening in the API)

File: app/models/snapshot.js (hunk 1 above)
Comment (cadeParade) [reply app/models/snapshot.js]: Yes, but it is not needed for this PR. This property also, is not used at all in this PR, it is copied from my import. I can remove it, but it's going to be added back soon.

File: tests/factories/comment-thread.js (hunk 2 above)
Comment (cadeParade) [reply tests/factories/comment-thread.js]: Sure.

**Pull request merged**